VirtoCommerce.ContentModule.Web/Controllers/Api/ContentController.cs

```diff
@@ -254,34 +266,31 @@ public IHttpActionResult CreateContentFolder(string contentType, string storeId,
                 {
                     remoteStream.CopyTo(blobStream);
 
-                    var retVal = new ContentFile
+                    retVal.Add(new ContentFile
                     {
                         Name = fileName,
                         Url = _urlResolver.GetAbsoluteUrl(fileUrl)
-                    };
-                    return Ok(retVal);
+                    });
                 }
             }
             else
             {
                 var blobMultipartProvider = new BlobStorageMultipartProvider(storageProvider, _urlResolver, folderUrl);
                 await Request.Content.ReadAsMultipartAsync(blobMultipartProvider);
 
-                var retVal = new List<ContentFile>();
-
-                foreach (var blobInfo in blobMultipartProvider.BlobInfos)
+                var files = blobMultipartProvider.BlobInfos.Select(blobInfo => new ContentFile
                 {
-                    retVal.Add(new ContentFile
-                    {
-                        Name = blobInfo.FileName,
-                        Url = _urlResolver.GetAbsoluteUrl(blobInfo.Key)
-                    });
-                }
-
-                return Ok(retVal.ToArray());
+                    Name = blobInfo.FileName,
+                    Url = _urlResolver.GetAbsoluteUrl(blobInfo.Key)
+                });
+                retVal.AddRange(files);
             }
+
+            _cacheManager.ClearRegion($"content-{storeId}");
```

Comment:
This line redundant because you use timeout expiration

Reply:
Clearing cache explicitly is needed in order to display a correct count after adding or deleting the pages. The system requests the statistics right after these operations, not after the cache expires.